### ISSUE TITLE
Rebuilding `HTTPError`s hides `response` and especially `response.status_code`

### ISSUE THREAD

**Issue by adamnovak** [ohsu-comp-bio/py-tes#33]
I want to catch `HTTPError`s from calls made with `py-tes`. Unfortunately, the exception generation code catches all `Exception`s, assumes them to have `HTTPError`-ish fields and turns them into `HTTPError`s with a custom message

https://github.com/ohsu-comp-bio/py-tes/blob/7606dc8c74fb69ac65e1fb3acf8c16aa4cb42d9e/tes/utils.py#L90-L97

Unfortunately, the `response` and `request` are not passed through to [Requests' constructor](https://docs.python-requests.org/en/master/_modules/requests/exceptions/#RequestException), and so when I go to `except HTTPError as e:` and then try to consult `e.response.status_code`, I get something like this:

```
  File "/home/anovak/workspace/toil/src/toil/batchSystems/tes.py", line 315, in __try_cancel
    if e.response.status_code == 500:
AttributeError: 'NoneType' object has no attribute 'status_code'
```

The custom error messages are nice, but I'd rather not have to try to parse the error codes out of the strings; is it possible to pass the request and response through when the error is reconstructed?

**Comment by kellrott** [comment]
Does #36 work?

**Comment by adamnovak** [comment]
This was fixed by #36.
